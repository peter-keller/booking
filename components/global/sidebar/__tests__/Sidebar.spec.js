import { shallowMount } from '@vue/test-utils'

import Sidebar from '..'

describe('Component: Sidebar', () => {
  const defaultOpts = {
    propsData: {
      name: 'navigation'
    }
  }

  const factory = (opts = {}) => shallowMount(Sidebar, {
    ...defaultOpts,
    ...opts
  })

  it('matches snapshot', () => {
    const wrapper = factory()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('matches snapshot when open', () => {
    const wrapper = factory({
      name: 'navigation',
      isOpen: true
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('matches snapshot with slots passed in', () => {
    const wrapper = factory({
      slots: {
        default: '<div>Test slot</div>'
      },
      propsData: {
        name: 'navigation',
        isOpen: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('$emits close event when close button is clicked', () => {
    const wrapper = factory({
      slots: {
        default: '<div>Test slot</div>'
      },
      propsData: {
        name: 'navigation',
        isOpen: true
      }
    })

    wrapper.find('icon-stub').vm.$emit('click')

    expect(wrapper.emitted('close')).toBeTruthy()
  })
})
