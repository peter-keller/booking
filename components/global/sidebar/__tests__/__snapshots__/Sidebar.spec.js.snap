// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Sidebar matches snapshot 1`] = `
"<div class=\\"sidebar\\">
  <!---->
  <transition-stub enterclass=\\"sidebar--enter\\" leavetoclass=\\"sidebar--leave-to\\" enteractiveclass=\\"sidebar--enter-active\\" leaveactiveclass=\\"sidebar--leave-active\\">
    <!---->
  </transition-stub>
</div>"
`;

exports[`Component: Sidebar matches snapshot when open 1`] = `
"<div class=\\"sidebar\\">
  <!---->
  <transition-stub enterclass=\\"sidebar--enter\\" leavetoclass=\\"sidebar--leave-to\\" enteractiveclass=\\"sidebar--enter-active\\" leaveactiveclass=\\"sidebar--leave-active\\">
    <!---->
  </transition-stub>
</div>"
`;

exports[`Component: Sidebar matches snapshot with slots passed in 1`] = `
"<div class=\\"sidebar\\">
  <div class=\\"sidebar__backdrop\\"></div>
  <transition-stub enterclass=\\"sidebar--enter\\" leavetoclass=\\"sidebar--leave-to\\" enteractiveclass=\\"sidebar--enter-active\\" leaveactiveclass=\\"sidebar--leave-active\\">
    <div class=\\"sidebar__panel\\">
      <div class=\\"sidebar__close\\">
        <icon-stub name=\\"close\\" size=\\"medium\\" stroke=\\"\\" clickable=\\"true\\"></icon-stub>
      </div>
      <div>Test slot</div>
    </div>
  </transition-stub>
</div>"
`;
